internal/services/storage/storage_account_resource.go

```diff
@@ -2149,22 +2149,26 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		blobClient := storageClient.BlobServicesClient
 		var blobProps storage.BlobServiceProperties
 
-		repeat := true
-		timeout := time.After(5 * time.Minute) // set timeout to 5 minutes
-		for repeat {
-			select {
-			case <-timeout:
-				log.Printf("[INFO] reading blob service properties timeout expired %s", *id)
-				return fmt.Errorf("reading blob service properties for %s: %+v", *id, err)
-			default:
+		// wait for blob service endpoint to become available
+
+		log.Printf("[INFO] reading %s blob service properties properties..", *id)
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"NotFound"},
+			Target:  []string{"Available"},
+			Refresh: func() (interface{}, string, error) {
 				blobProps, err = blobClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
 				if err != nil {
-					log.Printf("[INFO] refreshing blob service properties %s: %+v", *id, err)
-					time.Sleep(5 * time.Second) // wait for 5 seconds before retrying
-				} else {
-					repeat = false // no error, exit loop
+					log.Printf("[INFO] reading %s blob service properties properties: %+v", *id, err)
```

Comment:
Can we further check that the response type is 404 instead of consuming all error types?